### PR TITLE
feat: add tours

### PR DESCRIPTION
**Pull request by DavyJ0nes**
This introduces the concept of a tour to the Extend API. This is more closely aligned to the use cases that Saga supports today and is explained below.

This will replace the existing `Booking` message and its associated methods in the `BookingService`.

### What is a tour?

A Tour is a preplanned truck tour that includes what goods are to be picked up and delivered, where, at what times, and in what order.

When a tour is accepted and confirmed by Saga, Shipments based on the data provided in the Tour will be created.

### CODE REVIEW

File: proto/einride/saga/extend/book/v1beta1/tour.proto [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    // Service type drayage.
+    DRAYAGE = 5;
+  }
+
+  // The rule that is applied when a confirmed tour is accepted.
+  enum AutomationRule {
+    // Unspecified auto rule.
+    AUTOMATION_RULE_UNSPECIFIED = 0;
+    // No automation rule
+    NONE = 1;
```
Comment (gostajonasson) [inline proto/einride/saga/extend/book/v1beta1/tour.proto:106]: Did you say this should be removed?

File: proto/einride/saga/extend/book/v1beta1/tour.proto [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  // The rule that is applied when a confirmed tour is accepted. Defaults to CREATE_SHIPMENTS.
+  AutomationRule automation_rule = 11;
+
+  // The units included in this tour.
+  repeated Unit units = 12 [(google.api.field_behavior) = REQUIRED];
+  // The ordered list of stops included in this tour.
+  repeated Stop stops = 13 [(google.api.field_behavior) = REQUIRED];
+
+  // The shipments that are associated with the tour. These are created when a tour is CONFIRMED and ACCEPTED.
+  repeated Shipment shipments = 14 [(google.api.field_behavior) = OUTPUT_ONLY];
```
[the remark this answers is not in the document]
Comment (DavyJ0nes) [reply proto/einride/saga/extend/book/v1beta1/tour.proto:61]: that's correct, there's no issue with doing this and the same pattern is used in responses to Batch.* methods. The only rule that I can see is when having a field named unreachable https://google.aip.dev/217

File: proto/einride/saga/extend/book/v1beta1/booking_service.proto [hunk 3]
```diff
@@ -17,6 +19,7 @@ service BookingService {
   // Create a booking in a space.
   //
   // This is an AIP standard [Create](https://google.aip.dev/133) method.
+  // Deprecated: Use CreateTour instead.
```
Comment (DavyJ0nes) [inline proto/einride/saga/extend/book/v1beta1/booking_service.proto:22]: This was only added for the interim until we move internal customers over to Tours. These methods will then be removed from this API.

File: proto/einride/saga/extend/book/v1beta1/booking_service.proto [hunk 4]
```diff
@@ -48,6 +52,75 @@ service BookingService {
       }
     };
   }
+
+  // Create a truck tour booking in a space.
```
Comment (gostajonasson) [inline proto/einride/saga/extend/book/v1beta1/booking_service.proto:56]: Haven't seen "truck tour booking" before, is that the wording we should use?

File: proto/einride/saga/extend/book/v1beta1/tour.proto [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    STATE_UNSPECIFIED = 0;
+    // The booking is received. Awaiting accept/reject.
+    PENDING = 1;
+    // The booking is accepted.
+    ACCEPTED = 2;
+    // The booking is rejected.
+    REJECTED = 3;
+  }
+
+  // The service type of the booking.
```
Comment (gostajonasson) [inline proto/einride/saga/extend/book/v1beta1/tour.proto:85]:
```suggestion
  // The state of the tour.
  enum State {
    // Unspecified state
    STATE_UNSPECIFIED = 0;
    // The tour is received. Awaiting accept/reject.
    PENDING = 1;
    // The tour is accepted.
    ACCEPTED = 2;
    // The tour is rejected.
    REJECTED = 3;
  }

  // The service type of the tour.
```

File: proto/einride/saga/extend/book/v1beta1/booking_service.proto (hunk 4 above)
Comment (rikardq) [reply proto/einride/saga/extend/book/v1beta1/booking_service.proto:54]: We have vehicle in shipments :thinking:

**Review by rikardq — APPROVED** [review]
Great :star:

**Pull request merged**